### PR TITLE
feat: android security provider

### PR DESCRIPTION
**Pull request by dtarnawsky**

### CODE REVIEW

File: docs/mobile-security/android-security-provider.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+```bash
+npm install @capacitor-community/security-provider
+npx cap sync
+```
+
+In the startup of your application you should check if an update to the security provider is needed by adding the following code:
+```typescript
+import { CapacitorSecurityProvider, SecurityProviderStatus } from '@capacitor-community/security-provider';
+...
+const result = await CapacitorSecurityProvider.installIfNeeded();
```
Comment (jameson-w-taylor) [inline docs/mobile-security/android-security-provider.md:27]: Given the context later on about how this has no effect on iOS, maybe adjusting this snippet to wrap it in platform detection might simplify the conditional? Just personal preference so feel free to ignore this opinion.

Something like:
```ts
import { CapacitorSecurityProvider, SecurityProviderStatus } from '@capacitor-community/security-provider';

if (Capacitor.getPlatform() === 'android') {
  const { status } = await CapacitorSecurityProvider.installIfNeeded();
  if (status !== SecurityProviderStatus.Success) {
    // Do not proceed. The Android Security Provider failed to verify / install.
  }
}
```

File: docs/mobile-security/android-security-provider.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+---
+title: Android Security Provider
+sidebar_label: Android Security Provider
+sidebar_position: 4
+---
+
+import Admonition from '@theme/Admonition';
+
+Android relies on a security provider for secure network communications. However, from time to time, vulnerabilities are found in the default security provider. To protect against these vulnerabilities, Google Play services provides a way to automatically update a device's security provider to protect against known exploits. By calling Google Play services methods, you can help ensure that your app is running on a device that has the latest updates to protect against known exploits.
```
Comment (jameson-w-taylor) [inline docs/mobile-security/android-security-provider.md:9]: Maybe add link for "security provider":
>...relies on a security [Provider](https://developer.android.com/reference/java/security/Provider) for...

**Review by jameson-w-taylor — CHANGES_REQUESTED** [review]

File: docs/mobile-security/android-security-provider.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+The `status` property will return `SecurityProviderStatus.Success` on an Android device which is up to date. As this plugin has no effect on iOS the `status` property will return `SecurityProviderStatus.NotImplemented` on iOS devices.
+
+## Further Information
+
+Additional information about the Android security provider and the calls this plugin uses can be found in the [Android Developer Documentation](https://developer.android.com/privacy-and-security/security-gms-provider).
+
+## Summary
+
+Ensuring that your App verifies that you are using the latest secure version of Play Services is an additional step in ensuring the security of your Capacitor application.
```
[the remark this answers is not in the document]
Comment (dtarnawsky) [reply docs/mobile-security/android-security-provider.md]: Thanks, updated.

File: docs/mobile-security/android-security-provider.md (hunk 1 above)
Comment (dtarnawsky) [reply docs/mobile-security/android-security-provider.md:27]: The check for result.status !== SecurityProviderStatus.NotImplemented handles this for web and iOS. I guess a personal preference thing but I like the JS code base to have as few platform conditional as possible.

File: docs/mobile-security/android-security-provider.md (hunk 2 above)
Comment (dtarnawsky) [reply docs/mobile-security/android-security-provider.md]: Thanks, updated

**Review by jameson-w-taylor — APPROVED** [review]

File: docs/mobile-security/android-security-provider.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+You can implement the Google Play Services security provider check in a Capacitor application using the [`@capacitor-community/security-provider`](https://github.com/capacitor-community/android-security-provider) plugin.
+
+First, install the plugin:
+```bash
+npm install @capacitor-community/security-provider
+npx cap sync
+```
+
+In the startup of your application you should check if an update to the security provider is needed by adding the following code:
```
Comment (eric-horodyski) [inline docs/mobile-security/android-security-provider.md:23]: I would reword this sentence to flow better:

> In your application's startup code, check if an update to the security provider is needed by adding the following code:

**Review by eric-horodyski — APPROVED** [review]
One suggestion to make a sentence flow better, but not critical. Otherwise, LGTM.

**Pull request merged**